Fix scoped variables replacement in assetids such as repeat panels

ahom opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. To surface this PR in the changelog add the label: changelog
    If this PR is going in the changelog please make sure the title of the PR explains the feature in a user-centric way:
        Bad: fix state bug in hooks
        Good: Fix crash when switching from Query Builder

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, consider updating the documentation in README.md(https://github.com/grafana/iot-sitewise-datasource/blob/main/README.md) or [src/README.md](https://github.com/grafana/iot-sitewise-datasource/blob/main/README.md).

-->

**What this PR does / why we need it**:
This PR allows users to create repeat panels or rows when iterating over multiple assetIds. The repeat functionality was completely broken when trying to use it with the assetId property because ScopedVariables was not used during the replacement.

**Which issue(s) this PR fixes**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #178 

**Special notes for your reviewer**:
Previous implementation of the function was not taking into account the scopedVars that are needed for the repeat functionality within Grafana. I took the liberty to simplify the implementation, leveraging the 'csv' formatter in order to gracefully handle all cases (constant value, single-value and muti-value vars). This code won't work only if an assetId includes a comma, which I don't believe is valid.
I also removed the previous unit tests which are not relevant anymore as they were testing the function body with the mock of TemplateSrv. Happy to add any relevant test if there is anything that would be meaningful.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/iot-sitewise-datasource?pullRequest=205) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/iot-sitewise-datasource?pullRequest=205) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/iot-sitewise-datasource?pullRequest=205) it.</sub>

idastambuk commented:
Hi @ahom, thanks for the contribution! 
I tested it with repeating variables it looks good now. 
However, I wouldn't delete the old tests, as they do test previous bugs (just modify them, but keep the test cases). I think an additional one should be added for this fix as well, for scopedVars. Let us know if you need help with that.

ahom commented:
Thanks @idastambuk for the comments!
I added back the tests but had to modify the way they were working, I also added two new tests for ScopedVars as suggested.

idastambuk commented:
Hi @ahom sorry for the delay! Looks good, will merge this and release it
